internal/controller/state/graph/route_common.go

```diff
@@ -303,34 +303,36 @@ func buildSectionNameRefs(
 		}
 
 		gwNsName := client.ObjectKeyFromObject(gw.Source)
-		k := key{
-			gwNsName: gwNsName,
-		}
 
-		// If there is no section name, we create ParentRefs for each listener in the gateway
+		// If there is no section name, validate uniqueness by expanding in the uniqueness map only
 		if p.SectionName == nil {
+			// Check uniqueness for all listener names in this gateway
 			for _, l := range gw.Listeners {
-				k.sectionName = string(l.Source.Name)
+				k := key{
+					gwNsName:    gwNsName,
+					sectionName: string(l.Source.Name),
+				}
 
 				if err := checkUniqueSections(k); err != nil {
 					return nil, err
 				}
-
-				sectionNameRefs = append(sectionNameRefs, ParentRef{
-					// if the ParentRefs we create are for each listener in the same gateway, we keep the
-					// parentRefIndex the same so when we look at a route's parentRef's we can see
-					// if the parentRef is a unique parentRef or one we created internally
-					Idx:         i,
-					Gateway:     CreateParentRefGateway(gw),
-					SectionName: &l.Source.Name,
-					Port:        p.Port,
-				})
 			}
 
+			// Create a single parentRef without expansion
```

Comment:
same comment here about expansion, doesn't really make sense for someone looking at this code in the future